stores/bdbstore/badgerdb.go

```diff
@@ -129,6 +131,58 @@ func (db *Database) GetReadStatus(messageID mail.ID) (bool, error) {
 	return val[0] == 1, nil
 }
 
+func (db *Database) PutMessage(prefixKey string, message stores.Message) error {
+	mByte, err := json.Marshal(message)
+	if err != nil {
+		return err
+	}
+	return db.db.Update(func(txn *badger.Txn) error {
+		return txn.Set(db.messageKey(prefixKey, message.Headers.MessageID), mByte)
+	})
+}
+
+func (db *Database) GetMessages(prefixKey string) ([]stores.Message, error) {
+	var (
+		val []byte
+		err error
+	)
+
+	var messages []stores.Message
+
+	err = db.db.View(func(txn *badger.Txn) error {
+		iterator := txn.NewKeyIterator(db.messagePrefix(prefixKey), badger.DefaultIteratorOptions)
```

Comment:
Thats a good point. While I m putting the message, I m using `protocol/network/address.message.messageID` as a key. I think it's not involving the time in messageID field. I will come up with a much better key to provide order according to time.

Reply:
Badger is using `Lexicographic` order when we fetch it. [Source](https://github.com/dgraph-io/badger/issues/317). According to rfc2822, messageId must be unique but it does not force for a date to be used in it. Instead of including messageId, I think I can use `protocol/network/address.message.Headers.Date.Unix()` as a key. What do you think Rob? Do you have any other suggestions?

Reply:
The only other suggestion for ordering is using the block number. However this is not in the message as the message is created before the transaction is sent. I agree with your idea of using the date to maintain ordering, though as a unique key it doesn't see the correct fit. An option could be prepend an orderable element like time to the MessageID. This could give a best of both. e.g. Time + Separator + MessageID